fv3core/stencils/updatedzd.py

```diff
@@ -151,154 +256,144 @@ def out(zs: sd, zh: sd, ws: sd, dt: float):
             zh[0, 0, 0] = zh if zh > other else other
 
 
-def compute(ndif, damp_vtd, dp0, zs, zh, crx, cry, xfx, yfx, wsd, dt):
+def compute(
+    ndif: FloatField,
+    damp_vtd: FloatField,
+    dp0: FloatField,
+    zs: FloatField,
+    zh: FloatField,
+    crx: FloatField,
+    cry: FloatField,
+    xfx: FloatField,
+    yfx: FloatField,
+    wsd: FloatField,
+    dt: float,
+):
     grid = spec.grid
-
-    ndif[-1] = ndif[-2]
-    damp_vtd[-1] = damp_vtd[-2]
+    halo = grid.halo
 
     crx_adv = utils.make_storage_from_shape(
-        crx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        crx.shape, grid.compute_origin(add=(0, -halo, 0))
```

Comment:
alternatively can use the grid.full_origin and add an offset instead of subtract, but I don't think we've decided on a convention there, so either way is fine